src/Agent.Worker/JobExtension.cs

```diff
@@ -726,6 +727,32 @@ private void OutputSetupInfo(IExecutionContext context)
             }
         }
 
+        private void PublishKnobsInfo(IExecutionContext jobContext)
+        {
+            string jobId = jobContext?.Variables?.System_JobId?.ToString() ?? string.Empty;
+
+            var telemetryData = new Dictionary<string, string>()
+            {
+                { "JobId", jobId }
+            };
```

Comment:
Nit: it's probably better to post null instead of empty string as it's easier to filter later

Reply:
done